llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -2425,6 +2425,13 @@ def : GCNPat <(i1 imm:$imm),
   let WaveSizePredicate = isWave32;
 }
 
+let True16Predicate = UseRealTrue16Insts in
+foreach vt = [i32, v2i16] in
```

Comment:
I don't see how this pattern makes sense for the v2i16 case

Reply:
Hi Matt. I thought this works for any 16bit right shift on a VGPR_32. Why do you think this will not work for v2i16?

Reply:
A shift of 16 on an i32 takes the top half of the integer. A splat shift of 16 on a v2i16 is an out of bounds shift

Reply:
I might misunderstood srl on vector type. Let me try something and I will address this in the follow up patch

Reply:
Create a patch here https://github.com/llvm/llvm-project/pull/133458